### PR TITLE
Adds command line support for the maliput viz app.

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

Related to #15 

## Summary
Add CLI support for maliput_viz configuration.

The `maliput_viz` application supports passing a yaml file with the configuration to load a road network with a given backend.
The yaml file is expected to contain information about the maliput backend to use and the parameters to be passed to the selected backend.

https://user-images.githubusercontent.com/53065142/206535975-21401dbb-e73b-4c8d-b7ed-22d4c64a28d7.mp4

## Test it
Try:
```
maliput_viz --help
```
Create a yaml file with:
```yaml
---
maliput_viz:
  maliput_backend: "maliput_dragway"
  parameters:
    num_lanes: 16
    length: 1789
    lane_width: 3
    shoulder_width: 1
```
And then run
```
maliput_viz --yaml_file_path=<path-to-yaml-file>
```

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/maliput_viz/flags/gflags.cc [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  ss << "    The --yaml_file_path flag expects a valid path to a YAML file." << std::endl;
+  ss << "    The YAML file should contain the configuration for loading the road network." << std::endl;
+  ss << "    The structure of the YAML file should be: " << std::endl;
+  ss << "     |  maliput_viz:" << std::endl;
+  ss << "     |    maliput_backend: <backend_name> " << std::endl;
+  ss << "     |    parameters: " << std::endl;
+  ss << "     |       <key_1>: <value_1> " << std::endl;
+  ss << "     |       <key_2>: <value_2> " << std::endl;
+  ss << "     |       ...  " << std::endl;
+  ss << "     |       <key_N>: <value_N> " << std::endl << std::endl;
```
Comment (agalbachicar) [inline src/maliput_viz/flags/gflags.cc:94]: Why isn't it a string literal?

https://en.cppreference.com/w/cpp/language/string_literal

File: src/maliput_viz/flags/gflags.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+DECLARE_int32(verbosity);
+DECLARE_string(yaml_file_path);
+
+namespace maliput {
+namespace viz {
+namespace flags {
+
+void ParseCommandLineFlags(int argc, char** argv, bool remove_flags);
+
+void SetUsageMessage();
```
Comment (agalbachicar) [inline src/maliput_viz/flags/gflags.h:43]: I think you want to add a docstring.

File: src/maliput_viz/tools/yaml_parser.h [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+///      <key_2>: <value_2> "
+///      // ...
+///      <key_N>: <value_N> "
+/// @endcode
+class YamlConfigFileParser {
+ public:
+  /// @brief  Constructor
+  /// @param file_path Path to a YAML file.
+  YamlConfigFileParser(const std::string& file_path);
+  ~YamlConfigFileParser() = default;
```
Comment (agalbachicar) [inline src/maliput_viz/tools/yaml_parser.h:56]: Nit: consider omitting it.

File: src/maliput_viz/flags/gflags.cc [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    if (!path.exists() || !path.is_file()) {
+      std::cout << "Please verify that '" << value << "' is a valid file path." << std::endl;
+      return false;
+    }
+  }
+  return true;
+}
+
+DEFINE_validator(verbosity, &ValidateVerbosity);
+DEFINE_validator(yaml_file_path, &ValidateConfigFilePath);
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply src/maliput_viz/flags/gflags.cc:57]: https://github.com/maliput/maliput_integration/issues/126

File: src/maliput_viz/tools/yaml_parser.h [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+/// @code {.yaml}
+/// maliput_viz:"
+///   maliput_backend: <backend_name> "
+///   parameters: "
+///      <key_1>: <value_1> "
+///      <key_2>: <value_2> "
+///      // ...
+///      <key_N>: <value_N> "
+/// @endcode
+class YamlConfigFileParser {
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply src/maliput_viz/tools/yaml_parser.h]: Done.

File: src/maliput_viz/tools/yaml_parser.h (hunk 3 above)
Comment (francocipollone) [reply src/maliput_viz/tools/yaml_parser.h]: Done.

File: src/maliput_viz/flags/gflags.h [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <gflags/gflags.h>
+
+DECLARE_int32(verbosity);
+DECLARE_string(yaml_file_path);
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply src/maliput_viz/flags/gflags.h]: Yeah, I followed the recommendations [here](https://gflags.github.io/gflags/#declare). and using DECLARE_ made sense.

**However** I like your suggestion better. That way all `gflags` related is kept hidden in the implementation.

Done.

File: src/maliput_viz/flags/gflags.h (hunk 2 above)
Comment (francocipollone) [reply src/maliput_viz/flags/gflags.h]: Done.

File: src/maliput_viz/flags/gflags.h [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+namespace viz {
+namespace flags {
+
+void ParseCommandLineFlags(int argc, char** argv, bool remove_flags);
+
+void SetUsageMessage();
+
+}  // namespace flags
+}  // namespace viz
+}  // namespace maliput
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply src/maliput_viz/flags/gflags.h]: Done.

File: src/maliput_viz/flags/gflags.cc (hunk 1 above)
Comment (francocipollone) [reply src/maliput_viz/flags/gflags.cc]: Yeah, right.
Done.

File: src/maliput_viz/flags/gflags.cc [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+DEFINE_int32(verbosity, 3, "Verbosity is set within a range between [0-4]");
+DEFINE_string(yaml_file_path, "", "Path to the configuration YAML file");
+
+// Validators
+static bool ValidateVerbosity(const char* flagname, int32_t value) { return value >= 0 && value <= 4; }
+static bool ValidateConfigFilePath(const char* flagname, const std::string& value) {
+  if (!value.empty()) {
+    maliput::common::Path path{value};
+    if (!path.exists() || !path.is_file()) {
+      std::cout << "Please verify that '" << value << "' is a valid file path." << std::endl;
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply src/maliput_viz/flags/gflags.cc]: Indeed.
Done.

**Review by agalbachicar — APPROVED** [review]
LGTM

**Pull request merged**